### ISSUE TITLE
Requirement files do not correspond to extra dependencies

### ISSUE THREAD

**Issue by otaj** [Lightning-AI/lightning#15210]
### Bug description

`pip install -r requirements/pytorch/devel.txt` != `PACKAGE_NAME=pytorch pip install -e ".[dev]"`

Also the same holds for app.

cc @justusschock @Borda @carmocca 

### Error messages and logs

```

# Error messages and logs here please

```

**Comment by Borda** [comment]
yes, that is confusing, but it may not be fully aligned as requirements within the package or extra does not allow referring to source compared to requirements in text could; see #15285
So for example, `pip install -r requirements/pytorch/docs.txt != PACKAGE_NAME=pytorch pip install -e ".[docs]"` as we are using our template which is not published in PyPI

**Comment by otaj** [comment]
But we should be absolutely able to do that. `extras_require` allow everything `requirements.txt` file allows as well, including the link to our template.

I think `docs` themselves could be exempted from that if we don't want to allow to download our template, although that link is public anyway in the repo so I don't really see an issue in that

**Comment by carmocca** [comment]
Did you mean?:

`pip install -e . -r requirements/pytorch/devel.txt` != `PACKAGE_NAME=pytorch pip install -e ".[dev]"`

Becuase in the proposition of the top post, the left statement is not signaled to install the current package, so it's natural that they wouldn't be equal.

Maybe I am missing the point of this proposal :thinking:

**Comment by otaj** [comment]
Yes, that is more precise version of what I meant, you're right. My beef is with the fact that the list of requirements significantly differ, I took installing current package as a "prerequisite"

**Comment by Borda** [comment]
I think that we could update the requirements parsing to understand/follow `-r` command and append additional resources... that would solve the problems a clear way :otter: 
Just wondering if there is a tool that would do as are using `pkg_resources.parse_requirements`

**Comment by carmocca** [comment]
I have a stale PR improving this: https://github.com/Lightning-AI/lightning/pull/13912. Would need to be updated

**Comment by shenoynikhil** [comment]
I'd like to take this up!

**Comment by carmocca** [comment]
@shenoynikhil How do you plan to address this issue? There's no clear solution at the moment
